app/controllers/comments_controller.rb

```diff
@@ -1,20 +1,33 @@
 class CommentsController < ApplicationController
   before_action :authenticate_user!
```

Comment:
I think we need a before_action here that `require_current_comment`.  I think what's happening now is because it's `nil`, `require_current_comment_is_editable` is sneaking in and making it returning `unauthorized` instead of `not_found`.